### ISSUE TITLE
Targeted Python versions

### ISSUE THREAD

**Issue by ml-evs** [Matgenix/jobflow-remote#33]
Quick Q for @gpetretto, which Python version are we targeting with jobflow-remote? Jobflow itself is 3.8+, which is what we have been testing until recently, but there is quite a lot of 3.10+ code (just nicer type hints really). I guess given that the normal way of using jobflow-remote is inside the submission scripts, which share an environment with the real "science" code, we should try to be as forgiving as possible... If we want to stick with 3.8+ (i.e., just follow jobflow from now on) I can make a PR that "downgrades" the code.

**Comment by utf** [comment]
Now that numpy/scipy/matplotlib have all moved to Python 3.9+, my plan is transition jobflow in that direction too. Especially since we rely and monty and maggma which likely constrain us to Python 3.9+ in the future.

In terms of type hints, in case you're not aware, you can use:

```python
from __future__ import annotations
```

to get most of the nice features of Python 3.10 type hints.

**Comment by gpetretto** [comment]
Thanks @ml-evs for raising the point and @utf for providing some indications about the future plans. Even though I agree that in general it would be better to be forgiving in terms of requirements, if jobflow is already planning to drop support for 3.8 it is probably not worth spending time to make jobflow-remote compatible with it.

One note about the annotations. I think I have tried to use the `from __future__ import annotations` import in all the modules, but I could not use it in the modules where typer was used. I don't remember exactly what was the problem, but somehow avoiding the new annotation types and removing the future import fixed the issue. However I had only manually tested with python 3.9+. Is it in the `cli` module that python 3.8 fails? Or somewhere else?
